server/handlers/handlers.go

```diff
@@ -299,6 +299,12 @@ func (ah *ActivityHandler) handleCreatedActivity(activityIds msteams.ActivityIds
 		return discardedReasonOther
 	}
 
+	var timeElapsed float64
+	if newPost != nil {
+		timeElapsed = float64(time.Since(msg.CreateAt)) / float64(time.Second)
```

Comment:
Should we use `post.CreateAt` for consistency here? 

Also, this division doesn't make sense to me, since `time.Second = 1000 * 1000 * 1000``. If we're measuring seconds for this metric, we shouldn't need to divide at all. (But will need to do some calculations if we start using `post.CreateAt` as above.)

Reply:
We can't use `post.CreateAt` here as we are assigning the Mattermost post `createAt` time while creating a post [here](https://github.com/mattermost/mattermost-plugin-msteams-sync/blob/f34054a67779e0885fa9febe4e94c86914759e5b/server/handlers/converters.go#L63) the same as that of MS Teams posts `createAt` time to keep both the posts `createAt` time in sync.

The `time.Since()` returns the value in seconds, and by converting it to float64, we get a value in multiples of `10^9`. Similarly, `time.Second` returns value 1s, and by converting it to float64, we get it as `1e+9`. So we divide them to get the result in seconds.

Reply:
The only concern that I have here is if our server time is too much out of sync with the microsoft servers. Not a big concern but this measure is going to be a bit unreliable. I like the previous one because was measuring other thing, but in a more reliable way. We were measuring before how much time it takes to deliver the message since it hits our server until it his our database. So... that is helpful too. I see value having both metrics.

Reply:
> The time.Since() returns the value in seconds

Ah, right, I stumbled over `time.Since()` returning a duration which has to be converted back to seconds. Thanks for clarifying!

>  I like the previous one ...

We should pin down exactly what we want to measure with this metric. In my mind, the goal is to keep track of the "sync delay" or the timeliness of message delivery between Mattermost and MSTeams. In particular, I think we are interested in delays that originate from outside the plugin, e.g. networking delays or MSTeams falling behind. We have other metrics to track database performance which might influence when the enduser actually sees the post, but this graph would be most useful if we measure as close to the "network layer" as possible.

So for messages that originate from MSTeams, we really only have the timestamp that MSTeams gives us as well as `time.Now()`:

<img width="911" alt="image" src="https://github.com/mattermost/mattermost-plugin-msteams-sync/assets/1023171/5ae3e921-2522-473d-9ffe-131d9463ad44">

I think there's really nothing more to do that report `M1 - T0` which `time.Since` does nicely. Though one ask here is to record `time.Now` as early as possible in the request instead of after we've done any work. (Maybe even as early as the incoming HTTP request?)

For messages that originate from Mattermost, we can do something similar, but we have to measure 2x the value:

<img width="897" alt="image" src="https://github.com/mattermost/mattermost-plugin-msteams-sync/assets/1023171/39df3c16-e366-42a9-934b-16e795dfc3ac">

The ideal would be to measure `M2-M1` in this case, but of course we don't remember `M1`. So we stick with `M2-M0` as a compromise.

In practice, `M2-M0` for messages that originate from Mattermost should be roughly double `M1 - T0` for messages that originate from Teams. (There will be clock skew between MSTeams and Mattermost as well as the delta that is `M1-M0`.) So we might consider reporting `(M2-M0)/2` if we keep it in this metric. Alternatively, we could use separate metrics and report `sync_delay_seconds` and `round_trip_seconds` or some such.

Thoughts @jespino & @ayusht2810? Are there other ways of approaching this?

Reply:
@lieut-data this approach looks good to me, and I think we can move forward with it. I will start updating the code once @jespino agrees with the above approach as well.